### PR TITLE
fix bug related to single key when generating a report

### PR DESCRIPTION
**Pull request by lettergram**
If you attempt to generate a "pretty" report from the stripe-payments.csv it will throw an error.

```python
readable_report = profile.report(report_options={"output_format":"pretty"})
print(json.dumps(readable_report, indent=4))
```

The error:

```
Traceback (most recent call last):
  File "cprofile_test.py", line 121, in <module>
    print(json.dumps(human_readable_report, indent=4))
  File "/usr/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.7/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/lib/python3.7/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  [Previous line repeated 1 more time]
  File "/usr/lib/python3.7/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/usr/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type set is not JSON serializable

```

In the current code, you can you can add the omit_key: `data_stats.*.statistics.null_types_index` and the error wouldn't happen. 

```python
"omit_keys": ["data_stats.*.statistics.null_types_index"
readable_report = profile.report(
    report_options={
         "output_format":"pretty", 
         "omit_keys": ["data_stats.*.statistics.null_types_index"]
}
print(json.dumps(readable_report, indent=4))
```

There is a single dictionary entry for this file and I believe there are no values so it throws an error. 

Added a test and implemented a fix.

### CODE REVIEW

**Review by AnhTruong — COMMENTED** [review]
one comment

File: dataprofiler/profilers/profile_builder.py [hunk 1]
```diff
@@ -144,12 +144,17 @@ def profile(self):
             "column_name": self.name,
             "samples": self.sample,
         })
+        
+        null_index_list = {}
+        for null_type in self.null_types_index:
+            null_index_list[null_type] = list(self.null_types_index.keys())
```
[the remark this answers is not in the document]
Comment (lettergram) [reply dataprofiler/profilers/profile_builder.py:150]: Right now Null_index is of the form

{
    "nan": {53} 
}

We need it of the form:

{
    "nan": [53]
}

So we iterate through the keys and convert to a list.

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 2]
```diff
@@ -514,6 +515,23 @@ def test_correct_null_row_counts(self):
...
+        profiler_options = ProfilerOptions()
+        profiler_options.set({'data_labeler.is_enabled': False})
+        data = dp.Data(filename_null_in_file)
+        profile = dp.Profiler(data, profiler_options=profiler_options)
+
+        report = profile.report(report_options={"output_format":"pretty"})
+        
+        self.assertEqual(
+            report['data_stats']['invoice_id']['statistics']['null_types_index'],
+            {'': "['']"}
```
Comment (ChrisWallace2020) [inline dataprofiler/tests/profilers/test_profile_builder.py:531]: I am not familiar with null_types_index, but it says in the readme it maps a string to a list of ints, but this maps a string to a string (of a list of strings). So I feel like either this test case or the readme may be incorrect.

Comment (JGSweets) [reply dataprofiler/tests/profilers/test_profile_builder.py:531]: It seems the error is only with the report not making it serializable, should this instead go with all the report tests? Maybe it doesn't matter.

Comment (JGSweets) [reply dataprofiler/tests/profilers/test_profile_builder.py:531]: It does seem weird that the final value is an empty string for the list of values.

Comment (JGSweets) [reply dataprofiler/tests/profilers/test_profile_builder.py:531]: I believe the value should be: "[53]"

File: README.md [hunk 3]
```diff
@@ -245,23 +245,23 @@ pip3 install -r requirements-test.txt
...
 To run a file with Pytest use:
 ```
-pytest dataprofiler/tests/data_readers/test_csv_data.py -v
+DATAPROFILER_SEED=0 pytest dataprofiler/tests/data_readers/test_csv_data.py -v
 ```
 
 To run individual of unit test, use form:
 ```
-python3 -m unittest dataprofiler.tests.profilers.test_profile_builder.TestProfiler
+DATAPROFILER_SEED=0 python3 -m unittest dataprofiler.tests.profilers.test_profile_builder.TestProfiler
```
Comment (JGSweets) [inline README.md:264]: doesn't this need a `;` separating it?

File: README.md [hunk 4]
```diff
@@ -245,23 +245,23 @@ pip3 install -r requirements-test.txt
...
 
 ```
-python3 -m unittest discover -p test_profile_builder.py
+DATAPROFILER_SEED=0 python3 -m unittest discover -p test_profile_builder.py
 ```
 
 To run a file with Pytest use:
 ```
-pytest dataprofiler/tests/data_readers/test_csv_data.py -v
+DATAPROFILER_SEED=0 pytest dataprofiler/tests/data_readers/test_csv_data.py -v
```
Comment (JGSweets) [inline README.md:259]: doesn't this need a `;` separating it?

File: README.md [hunk 5]
```diff
@@ -245,23 +245,23 @@ pip3 install -r requirements-test.txt
...
 ```
-python3 -m unittest discover -p "test*.py"
+DATAPROFILER_SEED=0 python3 -m unittest discover -p "test*.py"
 ```
 
 To run file of unit tests, use form:
 
 ```
-python3 -m unittest discover -p test_profile_builder.py
+DATAPROFILER_SEED=0 python3 -m unittest discover -p test_profile_builder.py
```
Comment (JGSweets) [inline README.md:254]: doesn't this need a `;` separating it?

File: README.md [hunk 6]
```diff
@@ -245,23 +245,23 @@ pip3 install -r requirements-test.txt
 
 To run all unit tests, use:
 ```
-python3 -m unittest discover -p "test*.py"
+DATAPROFILER_SEED=0 python3 -m unittest discover -p "test*.py"
```
Comment (JGSweets) [inline README.md:248]: doesn't this need a `;` separating it?

**Review by JGSweets — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (lettergram) [reply README.md:248]: Not necessary, but you can. That technically creates a seperate execution

https://unix.stackexchange.com/questions/159489/is-there-a-difference-between-and-and

**Review by AnhTruong — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (lettergram) [reply README.md:259]: Not necessary, but you can. That technically creates a seperate execution

https://unix.stackexchange.com/questions/159489/is-there-a-difference-between-and-and

Comment (JGSweets) [reply README.md:259]: interesting, when i tested locally, that didnt work, curious why it isn't working on my machine

**Review by JGSweets — APPROVED** [review]

**Pull request merged**